Remove Quirk shouldStarBePermissionsPolicyDefaultValue for jsfiddle.net

karlcow opened the pull request:
#### 5f235ae331ec87577ec2decf92522d80d671b74b
<pre>
Remove Quirk shouldStarBePermissionsPolicyDefaultValue for jsfiddle.net
<a href="https://bugs.webkit.org/show_bug.cgi?id=284106">https://bugs.webkit.org/show_bug.cgi?id=284106</a>
<a href="https://rdar.apple.com/140975683">rdar://140975683</a>

Reviewed by Sihui Liu.

This removes shouldStarBePermissionsPolicyDefaultValue as it is not needed
after being tested.

* Source/WebCore/html/PermissionsPolicy.cpp:
(WebCore::parseAllowlist):
(WebCore::parsePolicyDirective):
(WebCore::PermissionsPolicy::processPermissionsPolicyAttribute):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldStarBePermissionsPolicyDefaultValue const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/QuirksData.h:

Canonical link: <a href="https://commits.webkit.org/287433@main">https://commits.webkit.org/287433@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6f3dec7f014bb7899afed3b5a10783ca96bb2f67

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/79666 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58662 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/33057 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/84205 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30702 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/81775 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/67750 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/6951 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/62268 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/20122 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/82732 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/52324 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/72545 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42576 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/49667 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/26697 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/29133 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/70807 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/27156 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/85621 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/6911 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/4820 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/70519 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/7080 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/68391 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69765 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17395 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13778 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/12684 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/6863 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/12499 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/6748 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/10243 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8547 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/html/PermissionsPolicy.cpp:
```diff
@@ -367,7 +367,7 @@ static PermissionsPolicy::PolicyDirective parsePolicyDirective(StringView value,
 {
     auto allowAttributeValue = iframe.attributeWithoutSynchronization(allowAttr);
-    auto policyDirective = parsePolicyDirective(allowAttributeValue, iframe.document().securityOrigin().data(), declaredOrigin(iframe)->data(), iframe.document().quirks().shouldStarBePermissionsPolicyDefaultValue());
+    auto policyDirective = parsePolicyDirective(allowAttributeValue, iframe.document().securityOrigin().data(), declaredOrigin(iframe)->data(), false);
```
Since this is the only caller of this algorithm as far as I can tell, you should really update the algorithm to remove the argument and its associated branching altogether.

stwrt replied:
There are two functions that should be updated `parsePolicyDirective` and `parseAllowlist`. Both functions are only called in one place.

karlcow replied:
I removed the true/false parameters and if check

szewai commented on Source/WebCore/html/PermissionsPolicy.cpp:
```diff
@@ -310,12 +310,9 @@ static bool computeFeatureEnabled(PermissionsPolicy::Feature feature, const Docu
-static Allowlist parseAllowlist(StringView value, const SecurityOriginData& containerOrigin, const SecurityOriginData& targetOrigin, bool useStarAsDefaultAllowlistValue)
+static Allowlist parseAllowlist(StringView value, const SecurityOriginData& containerOrigin, const SecurityOriginData& targetOrigin)
 {
     if (value.isEmpty()) {
```
No need for `{`

karlcow replied:
🤦‍♂️ Thanks!

karlcow committed:
Remove Quirk shouldStarBePermissionsPolicyDefaultValue for jsfiddle.net
https://bugs.webkit.org/show_bug.cgi?id=284106
rdar://140975683

Reviewed by Sihui Liu.

This removes shouldStarBePermissionsPolicyDefaultValue as it is not needed
after being tested.

* Source/WebCore/html/PermissionsPolicy.cpp:
(WebCore::parseAllowlist):
(WebCore::parsePolicyDirective):
(WebCore::PermissionsPolicy::processPermissionsPolicyAttribute):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldStarBePermissionsPolicyDefaultValue const): Deleted.
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/QuirksData.h:

Canonical link: https://commits.webkit.org/287433@main
